cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java

```diff
@@ -522,52 +383,21 @@ private Resource<Kafka> kafkaAssembly(String namespace, String name) {
-        context.verify(() -> {
-            assertThat(statefulSet, is(notNullValue()));
-            // Check the storage class is initially "foo"
-            List<PersistentVolumeClaim> volumeClaimTemplates = statefulSet.getSpec().getVolumeClaimTemplates();
-            assertThat(volumeClaimTemplates, hasSize(1));
-            assertThat(volumeClaimTemplates.get(0).getSpec().getStorageClassName(), is(expectedClass));
-        });
-    }
-
-    @ParameterizedTest
-    @MethodSource("data")
-    public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext context) {
-        init(params);
+// Doubt in this method. Particularly stuck on how to get the Volumes and the init container using the PodSet
 
-        AtomicReference<List<PersistentVolumeClaim>> originalPVCs = new AtomicReference<>();
-        AtomicReference<List<Volume>> originalVolumes = new AtomicReference<>();
-        AtomicReference<List<Container>> originalInitContainers = new AtomicReference<>();
 
+    @Test
+    public void testReconcileUpdatesKafkaStorageType(VertxTestContext context) {
         Checkpoint async = context.checkpoint();
         initialReconcile(context)
             .onComplete(context.succeeding(v -> context.verify(() -> {
-                originalPVCs.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
-                        .map(StatefulSet::getSpec)
-                        .map(StatefulSetSpec::getVolumeClaimTemplates)
-                        .orElse(new ArrayList<>()));
-                originalVolumes.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
-                        .map(StatefulSet::getSpec)
-                        .map(StatefulSetSpec::getTemplate)
-                        .map(PodTemplateSpec::getSpec)
-                        .map(PodSpec::getVolumes)
-                        .orElse(new ArrayList<>()));
-                originalInitContainers.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
-                        .map(StatefulSet::getSpec)
-                        .map(StatefulSetSpec::getTemplate)
-                        .map(PodTemplateSpec::getSpec)
-                        .map(PodSpec::getInitContainers)
-                        .orElse(new ArrayList<>()));
-
                 // Update the storage type
                 // ephemeral -> persistent
                 // or
                 // persistent -> ephemeral
                 Kafka updatedStorageKafka = null;
-                if (kafkaStorage instanceof EphemeralStorage) {
+                System.out.println(cluster.getSpec().getKafka().getStorage().getClass());
+                if (cluster.getSpec().getKafka().getStorage() instanceof EphemeralStorage) {
```

Comment:
You seem to always start with persistent storage. So why is this needed? It seems like it doesn't do anything.